buf/registry/owner/v1beta1/user_service.proto

```diff
@@ -82,6 +82,23 @@ message ListUsersRequest {
   // If this is empty, all Users for all Organizations are listed, but this functionality
   // is limited to Users with the necessary permissions.
   repeated OrganizationRef organization_refs = 3;
+  message Filter {
+    // Filter by the type of the User.
+    repeated UserType type = 1 [(buf.validate.field).enum.defined_only = true];
+    // Filter by the state of the User.
+    repeated UserState state = 2 [(buf.validate.field).enum.defined_only = true];
+  }
+  // The filter to apply.
+  Filter filter = 4;
+  enum OrderDirection {
+    ORDER_DIRECTION_UNSPECIFIED = 0;
+    ORDER_DIRECTION_ASC = 1;
+    ORDER_DIRECTION_DESC = 2;
+  }
+  // The order to sort the Users.
+  //
+  // If empty, the default ORDER_DIRECTION_ASC is used.
+  OrderDirection order_direction = 5 [(buf.validate.field).enum.defined_only = true];
```

Comment:
`order_direction` over `order` to be able to potentially add `order_by` later if needed. Maybe `order` is simpler.